jobbergate-api/jobbergate_api/tests/apps/job_submissions/test_properties_parser.py

```diff
@@ -0,0 +1,656 @@
+import contextlib
+import inspect
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import MutableMapping
+from unittest import mock
+from pydantic import ValidationError
+
+import pytest
+from bidict import bidict
+
+from jobbergate_api.apps.job_scripts.job_script_files import JobScriptFiles
+from jobbergate_api.apps.job_submissions.properties_parser import (
+    _IDENTIFICATION_FLAG,
+    _INLINE_COMMENT_MARK,
+    ArgumentParserCustomExit,
+    _clean_jobscript,
+    _clean_line,
+    _flagged_line,
+    _split_line,
+    build_mapping_sbatch_to_slurm,
+    build_parser,
+    convert_sbatch_to_slurm_api,
+    get_job_parameters,
+    get_job_properties_from_job_script,
+    jobscript_to_dict,
+    sbatch_to_slurm,
+)
+from jobbergate_api.apps.job_submissions.schemas import JobProperties
+
+
+def test_identification_flag():
+    """
+    Check if the value of the identification flag is the same codded in the tests.
+    """
+    assert _IDENTIFICATION_FLAG == "#SBATCH"
+
+
+def test_inline_comment_mark():
+    """
+    Check if the inline comment mark is the same codded in the tests.
+    """
+    assert _INLINE_COMMENT_MARK == "#"
+
+
+@pytest.mark.parametrize(
+    "line, desired_value",
+    (
+        ("#SBATCH -abc", True),
+        ("##SBATCH -abc", False),
+        ("run python application", False),
+        ("# A comment", False),
+    ),
+)
+def test_flagged_line(line, desired_value):
+    """
+    Check if the flagged lines are identified properly.
+    """
+    actual_value = _flagged_line(line)
+    assert actual_value == desired_value
+
+
+@pytest.mark.parametrize(
+    "line, desired_value",
+    (
+        ("#SBATCH", ""),
+        ("#SBATCH#SBATCH", ""),
+        ("#SBATCH -abc # A comment", "-abc"),
+        ("#SBATCH --abc=0 # A comment", "--abc=0"),
+        ("#SBATCH --abc 0 # A comment", "--abc 0"),
+        ("#SBATCH -a 0 # A comment", "-a 0"),
+        ("#SBATCH -a=0 # A comment", "-a=0"),
+        ("#SBATCH -a = 0 # A comment", "-a = 0"),
+        ("#SBATCH -a=C&Aaccount # A comment", "-a=C&Aaccount"),
+    ),
+)
+def test_clean_line(line, desired_value):
+    """
+    Check if the provided lines are cleaned properly, i.e.,
+    identification flag, inline comments, and comment mark are all removed.
+    """
+    actual_value = _clean_line(line)
+    assert actual_value == desired_value
+
+
+@pytest.mark.parametrize(
+    "line, desired_value",
+    (
+        ("--help", ["--help"]),
+        ("--abc=0", ["--abc", "0"]),
+        ("-J job_name", ["-J", "job_name"]),
+        ("-v -J job_name", ["-v", "-J", "job_name"]),
+        ("-J job_name -v", ["-J", "job_name", "-v"]),
+        ("-a=0", ["-a", "0"]),
+        ("-a = 0", ["-a", "0"]),
+        ("-a 0", ["-a", "0"]),
```

Comment:
so, I was running some tests locally and I noticed slurm only accepts the `=` sign when using flags with `--`. When the flag is set with `-`, slurm accepts a white space between the argument and the value. 

In case you try to pass `=` when you should be using white space, slurm will point out an error. Example
<img width="856" alt="image" src="https://user-images.githubusercontent.com/45343359/205949582-fca9837a-8f25-481a-b373-35fa70cab439.png">

I think we should perhaps adjust this parser to only accept what slurm accepts. Thoughts?

Reply:
don't you think it is inconsistent to accept parameters that slurm itself doesn't?

It can drive users to run jobs by jobbergate that won't be run by slurm. Thinking as an engineer who might want to use jobbergate, I think it is very inconsistent and strange such a behaviour

Reply:
Ok, I agree it should be investigated in fixed, but not on this PR since the syntax as expected by `srun` is already covered. In fact, the syntax here should not be more permissive than that on slurm. I updated the Technical notes on [PFT-631](https://app.clickup.com/t/18022949/PFT-631) and moved it to Priority Backlog.

Reply:
perfect, thank you